float32[] data
---
int8 index